src/server/game/Handlers/MovementHandler.cpp

```diff
@@ -204,26 +205,63 @@ void WorldSession::HandleMoveWorldportAck()
 void WorldSession::HandleMoveTeleportAck(WorldPacket& recvData)
 {
     TC_LOG_DEBUG("network", "MSG_MOVE_TELEPORT_ACK");
-    ObjectGuid guid;
 
+    Unit* mover = _player->GetUnitBeingMoved();
+    ASSERT(mover);
```

Comment:
Is this ever possible to trigger under any conditions? If not then I would like to see that ASSERT placed in code setting this
Otherwise asserts arent ever a good idea in client handlers

Reply:
Honestly I don't know. IIRC, that ASSERT was already there before I started refactoring. Maybe it should be removed.

'I would like to see that ASSERT placed in code setting this'
What did you mean by 'in code setting this' ? In the body of GetUnitBeingMoved() you mean?